File: MyLeetCodeSolutions/PythonSolutions/Solution121.py
#121. Best Time to Buy and Sell Stock
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        difference, current, min = 0, 0, prices[0]
        for i in range(0, len(prices)):
            if prices[i] < min:
                min = prices[i]
            current = prices[i] - min
            if difference < current:
                difference = current
        return difference

File: 2022-2023/MyLeetCodeSolutions/JavaSolutions/Solution121.java
//121. Best Time to Buy and Sell Stock
class Solution {
    public int maxProfit(int[] prices) {
        int difference = 0;
        int current = 0;
        int min = prices[0];

        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < min) {
                min = prices[i];
            }
            current = prices[i] - min;
            if (difference < current) {
                difference = current;
            }
        }
        return difference;
    }
}
